core/src/main/java/fr/sncf/osrd/envelope/Envelope.java

```diff
@@ -151,7 +161,12 @@ public double interpolateSpeed(double position) {
     public long interpolateTotalTimeMS(double position) {
         assert continuous : "interpolating times on a non continuous envelope is a risky business";
         var envelopePartIndex = findEnvelopePartIndex(position);
-        assert envelopePartIndex != -1;
+        if (envelopePartIndex == -1) {
+            if (position < getEndPos() + 1e-8)
```

Comment:
I trust your bug fix here but it's a bit worrying that we call this method interpolateTotalTimeMS after the end of the path.

Reply:
We did something that ended up (mathematically) equivalent to `envelope.interpolateTotalTime(envelope.getEndPos())`, but we ended up with something like a 1e-15 difference. I assumed float inaccuracies and didn't look further.

Reply:
@eckter our policy so far (with alex) has been to track down these things to avoid hack like these everywhere. we embrace inaccuracies but strive to preserve basic properties (such as upper / lower bounds)

Reply:
I tried to reproduce it to see where exactly it comes from, but I couldn't manage to make it happen again... It's very weird. I'll remove it.